refurb/checks/pathlib/getsize.py

```diff
@@ -0,0 +1,77 @@
+from dataclasses import dataclass
+
+from mypy.nodes import BytesExpr, CallExpr, MemberExpr, NameExpr, StrExpr, Var
+
+from refurb.checks.pathlib.util import is_pathlike
+from refurb.error import Error
+
+
+@dataclass
+class ErrorInfo(Error):
+    """
+    Don't use the `os.path.getsize` (or similar) functions, use the more modern
+    `pathlib` module instead:
+
+    Bad:
+
+    ```
+    if os.path.getsize("file.txt"):
+        pass
+    ```
+
+    Good:
+
+    ```
+    if Path("file.txt").stat().st_size:
+        pass
+    ```
+    """
+
+    name = "use-pathlib-stat"
+    code = 155
+    categories = ["pathlib"]
+
+
+PATH_TO_PATHLIB_NAMES = {
+    "posixpath.getsize": "stat().st_size",
+    "genericpath.getsize": "stat().st_size",
+    "ntpath.getsize": "stat().st_size",
+    "os.stat": "stat()",
+    "os.path.getatime": "stat().st_atime",
+    "os.path.getmtime": "stat().st_mtime",
+    "os.path.getctime": "stat().st_ctime",
+
+}
+
+
+def check(node: CallExpr, errors: list[Error]) -> None:
+    match node:
+        case CallExpr(
+            callee=MemberExpr(fullname=fullname, name=name),
+            args=[arg],
+        ) if new_name := PATH_TO_PATHLIB_NAMES.get(fullname or ""):
```

Comment:
See my review comment above for context.

I would suggest changing this section to the following:

```python
PATH_TO_PATHLIB_NAMES = {
    "os.stat": "stat()",
    "os.path.getsize": "stat().st_size",
    "os.path.getatime": "stat().st_atime",
    "os.path.getmtime": "stat().st_mtime",
    "os.path.getctime": "stat().st_ctime",
}


def check(node: CallExpr, errors: list[Error]) -> None:
    match node:
        case CallExpr(
            callee=MemberExpr(fullname=fullname),
            args=[arg],
        ):
            normalized_name = fullname or ""

            for name in ("posix", "nt", "generic"):
                if normalized_name.startswith(name):
                    normalized_name = normalized_name.replace(name, "os.")

            new_name = PATH_TO_PATHLIB_NAMES.get(normalized_name)

            if not new_name:
                return
```

This basically will normalize the `posixpath`, `genericpath` and `ntpath` fields to `os.path`. This makes the lookup logic simpler and more self-explanatory.

Reply:
thank you